src/DataStructures/DataVector.hpp

```diff
@@ -70,15 +70,29 @@ using std::abs;  // NOLINT
 
 /*!
  * \ingroup DataStructuresGroup
- * \brief A class for storing data on a mesh.
+ * \brief Stores a collection of function values.
  *
- * A Data holds an array of contiguous data and can be either owning (the array
- * is deleted when the Data goes out of scope) or non-owning, meaning it just
- * has a pointer to an array.
+ * \details Use DataVector to represent function values on the computational
+ * domain. Note that interpreting the data also requires knowledge of the points
+ * that these function values correspond to.
  *
- * A variety of mathematical operations are supported with DataVectors. In
- * addition to the addition, subtraction, multiplication, division, etc. there
- * are the following element-wise operations:
+ * A DataVector holds an array of contiguous data. The DataVector can be owning,
+ * meaning the array is deleted when the DataVector goes out of scope, or
+ * non-owning, meaning it just has a pointer to an array.
+ *
+ * Unless you have a reason to use DataVector directly, you should generally use
```

Comment:
This is not true. Temporaries inside functions are fine to be `DataVector` but any arguments and return types should be `Scalar<DataVector>`